### PR TITLE
Integration Of SOMEIP to PCAP Network Trace

### PR DESCRIPTION
**Pull request by Prasanna-Vadanan**
Added a new protocol to the Network Trace feature of Trace Compass

### CODE REVIEW

File: pcap/org.eclipse.tracecompass.pcap.core/src/org/eclipse/tracecompass/internal/pcap/core/protocol/someip_service_discovery/SOMEIP_SDEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2014 Ericsson
```
Comment (tahini) [inline pcap/org.eclipse.tracecompass.pcap.core/src/org/eclipse/tracecompass/internal/pcap/core/protocol/someip_service_discovery/SOMEIP_SDEndpoint.java]: Careful with copyright headers, this should be your organization and 2018

File: pcap/org.eclipse.tracecompass.pcap.core/src/org/eclipse/tracecompass/internal/pcap/core/protocol/someip_service_discovery/SOMEIP_SDEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.tracecompass.internal.pcap.core.endpoint.ProtocolEndpoint;
+import org.eclipse.tracecompass.internal.pcap.core.protocol.udp.UDPEndpoint;
+
+/**
+ * Class that extends the ProtocolEndpoint class. It represents the endpoint at
+ * an UDP level.
+ *
+ * @author Vincent Perot
```
Comment (tahini) [inline pcap/org.eclipse.tracecompass.pcap.core/src/org/eclipse/tracecompass/internal/pcap/core/protocol/someip_service_discovery/SOMEIP_SDEndpoint.java]: And here you should have your name

File: pcap/org.eclipse.tracecompass.tmf.pcap.core/src/org/eclipse/tracecompass/internal/tmf/pcap/core/protocol/TmfPcapProtocol.java [hunk 3]
```diff
@@ -66,7 +67,15 @@
      * a "payload packet". This is considered to be on layer 7 since its always
      * the most encapsulated packet if present.
      */
-    UNKNOWN(PcapProtocol.UNKNOWN);
+    UNKNOWN(PcapProtocol.UNKNOWN),
```
Comment (tahini) [inline pcap/org.eclipse.tracecompass.tmf.pcap.core/src/org/eclipse/tracecompass/internal/tmf/pcap/core/protocol/TmfPcapProtocol.java]: Leave unknown as the last one

File: pcap/org.eclipse.tracecompass.tmf.pcap.core/src/org/eclipse/tracecompass/internal/tmf/pcap/core/util/ProtocolConversion.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2014 Ericsson
+ *
```
Comment (tahini) [inline pcap/org.eclipse.tracecompass.tmf.pcap.core/src/org/eclipse/tracecompass/internal/tmf/pcap/core/util/ProtocolConversion.java]: Do not remove copyright headers from these files, you can put 2014, 2018 Ericsson, your organization

**Review by tahini — CHANGES_REQUESTED** [review]
Thank you for this pull request. I did a basic review, mostly of copyright stuff.

But the github repository is only a mirror of the official Trace Compass repository and it is not where we review and accept patches. Here are the instructions to submit patches through Eclipse's gerrit: https://wiki.eclipse.org/Trace_Compass/Contributor_Guidelines

You should squash your commits into 1 and push it on gerrit as per instructions above. We will do a more thorough review once on gerrit.

Also please provide a trace, that you can add in the unit test plugin so we can test the new protocol. That would help review the patch.

**Comment by TheMatthew** [comment]
Hey, I would like to counter, please push the branch as-is to gerrit, it should be easier to commit. I like the raviolitude of the patches. ;)

**Comment by TheMatthew** [comment]
Went through the entire patchset... I agree with Genevieve, the whole branch chain should be squashed as it does not compile in some parts.

**Comment by bhufmann** [comment]
Closing this one. This mirror will be archived. If PR is still relevant please contribute it to: https://github.com/eclipse-tracecompass/org.eclipse.tracecompass. Thanks!

**Pull request closed without merging**